core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -340,9 +338,9 @@ public B addHttpHeader(CharSequence name, Object value) {
     /**
      * Adds the specified HTTP headers.
      */
-    public B addHttpHeaders(Headers<AsciiString, String, ?> httpHeaders) {
+    public B addHttpHeaders(Iterable<? extends Entry<? extends CharSequence, ?>> httpHeaders) {
```

Comment:
Is there any rationale to move from AsciiString, String to CharSequence ? is that just loosen the spec that HTTP header only able to be US-ASCII?

Reply:
Internally, when you specify a `CharSequence` as a name, it will be automatically converted to an `AsciiString` using `HttpHeaderNames.of()`. The change is purely for user's convenience:

```java
// Before:
HttpHeaders headers = HttpHeaders.of(HttpHeaderNames.of("x-foo", "bar");
// After:
HttpHeaders headers = HttpHeaders.of("x-foo", "bar");
```

Unlike the names, the values are stored as-is in `CharSequence`, and converted to a `String` lazily. This change is based on the assumption that a user does not read all header values in reality, hence reducing the cost of unnecessary `CharSequence/AsciiString` to `String` conversion.